R/corridor.R

```diff
@@ -283,5 +301,17 @@ cap_corridor <- function(edges, method = "direct", network = NULL) {
       sprintf("Unknown method to cap the river corridor: {method}", method)
     )
   }
-  as_polygon(c(edges, cap_edge_1, cap_edge_2))
+  polygon <- as_polygon(c(edges, cap_edge_1, cap_edge_2))
+
+  # If the capping edges intersect the given corridor edges in points other
+  # than the end points, the polygonization of the corridor boundary leads to
+  # small side polygons. We drop these, after raising a warning
+  if (length(polygon) > 1) {
+    warning(
+      "Corridor capping gives multiple polygons - selecting the largest one"
+    )
+    polygon <- polygon[find_largest(polygon)]
+  }
+
+  polygon
```

Comment:
@cforgaci  Following up on https://github.com/CityRiverSpaces/CRiSp/pull/96#discussion_r1995857151 , do you this is clear now? Moved it inside `cap_corridor` since this is where the problem can actually arise.

Reply:
Yes, super clear now!